src/main/java/betterquesting/api2/client/gui/panels/lists/CanvasQuestSearch.java

```diff
@@ -0,0 +1,48 @@
+package betterquesting.api2.client.gui.panels.lists;
+
+import betterquesting.api.properties.NativeProps;
+import betterquesting.api.questing.IQuest;
+import betterquesting.api.questing.IQuestLine;
+import betterquesting.api2.client.gui.misc.IGuiRect;
+import betterquesting.api2.client.gui.panels.lists.CanvasSearch;
+import betterquesting.api2.storage.DBEntry;
+import betterquesting.api2.utils.QuestTranslation;
+import betterquesting.misc.QuestSearchEntry;
+import betterquesting.questing.QuestDatabase;
+import betterquesting.questing.QuestLineDatabase;
+
+import java.util.ArrayDeque;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public abstract class CanvasQuestSearch extends CanvasSearch<QuestSearchEntry, QuestSearchEntry> {
+    private List<QuestSearchEntry> questList;
+
+    public CanvasQuestSearch(IGuiRect rect) {
+        super(rect);
+    }
+
+    @Override
+    protected Iterator<QuestSearchEntry> getIterator() {
+        if (questList == null){
```

Comment:
invert the if here please, 
``
if (questList != null)
    return questList.iterator();
``
then continue your regular code. this reduces nesting